dts/arm/st/u5/stm32u535.dtsi

```diff
@@ -0,0 +1,37 @@
+/*
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <st/u5/stm32u5.dtsi>
+
+/ {
+  soc {
+    /* USB-C PD is not available on this part. */
+    /delete-node/ ucpd@4000dc00;
+
+    /* Advanced Encryption Standard hardware accelerator is not available on this part. */
+    /delete-node/ aes@420c0000;

```

Comment:
In theory, we prefer to minimize use of /delete-node/ but looking to current set of files a good clean up is required in the whole U5 .dtsi files to better reflect the series tree which is depicted here:
![image](https://github.com/user-attachments/assets/65639215-83e6-4a7f-b46a-6bc6761aae72)

So, let's keep if that way, but it will be refactored soon.

Reply:
Created https://github.com/zephyrproject-rtos/zephyr/issues/89704 to address this issue.